### PR TITLE
[Clang][NFC] Remove unnecessary copy

### PR DESCRIPTION
**Pull request by smanna12**
Reported by Static Analyzer Tool:

In clang::ASTNodeImporter::VisitCountAttributedType(clang::CountAttributedType const *): Using the auto keyword without an & causes the copy of an object of type TypeCoupledDeclRefInfo

### CODE REVIEW

File: clang/lib/AST/ASTImporter.cpp [hunk 1]
```diff
@@ -1551,7 +1551,7 @@ ASTNodeImporter::VisitCountAttributedType(const CountAttributedType *T) {
   Expr *CountExpr = importChecked(Err, T->getCountExpr());
 
   SmallVector<TypeCoupledDeclRefInfo, 1> CoupledDecls;
-  for (auto TI : T->dependent_decls()) {
+  for (const TypeCoupledDeclRefInfo &TI : T->dependent_decls()) {
```
Comment (JOE1994) [inline clang/lib/AST/ASTImporter.cpp:1554]:
```suggestion
  for (const auto &TI : T->dependent_decls()) {
```

Comment (smanna12) [reply clang/lib/AST/ASTImporter.cpp]: Thank you @JOE1994 for reviews. I have updated patch.

**Review by cor3ntin — APPROVED** [review]

**Comment by smanna12** [comment]
Thank you, @cor3ntin for reviews!

File: clang/lib/AST/ASTImporter.cpp (hunk 1 above)
Comment (rapidsna) [reply clang/lib/AST/ASTImporter.cpp]: Was there a reason to suggest it back to `auto` from `TypeCoupledDeclRefInfo`? I thought Clang prefers to have an explicit type name when the deduced type is not explicit from RHS.

**Review by rapidsna — APPROVED** [review]
Thanks for fixing this issue! I left a minor comment. Other than that, LGTM.

File: clang/lib/AST/ASTImporter.cpp (hunk 1 above)
Comment (smanna12) [reply clang/lib/AST/ASTImporter.cpp:1554]: Done. Thanks for the review's !

Comment (MaskRay) [reply clang/lib/AST/ASTImporter.cpp:1554]: I agree `TypeCoupledDeclRefInfo` should be kept. The `auto` here is not clear.

**Review by MaskRay — APPROVED** [review]
Can you add a NFC tag to the subject? (patch subject; commit message are completely ignored since we use "Squash and merge")

**Comment by smanna12** [comment]
Thank you everyone for reviews!

**Pull request merged**